typescript/infra/scripts/funding/utils/grafana.ts

```diff
@@ -0,0 +1,167 @@
+import { ChainMap } from '@hyperlane-xyz/sdk';
+import { rootLogger } from '@hyperlane-xyz/utils';
+
+import { fetchGCPSecret } from '../../../src/utils/gcloud.js';
+
+import { BalanceThresholdConfig, configFileNameMapping } from './constants.js';
+
+export const GRAFANA_URL = 'https://abacusworks.grafana.net';
+
+export const THRESHOLD_CONFIG_PATH = './config/environments/mainnet3/balances';
+
+export const logger = rootLogger.child({ module: 'grafana' });
+
+export enum AlertType {
+  LowUrgencyKeyFunderBalance = 'lowUrgencyKeyFunderBalance',
+  LowUrgencyEngKeyFunderBalance = 'lowUrgencyEngKeyFunderBalance',
+  HighUrgencyRelayerBalance = 'highUrgencyRelayerBalance',
+}
+
+export type AlertTypeValue = `${AlertType}`;
+
+export enum WalletName {
+  KeyFunder = 'keyFunder',
+  Relayer = 'relayer',
+}
+
+export type WalletNameValue = `${WalletName}`;
+
+const alertWalletNameMapping: Record<AlertTypeValue, WalletName> = {
+  [AlertType.LowUrgencyKeyFunderBalance]: WalletName.KeyFunder,
+  [AlertType.LowUrgencyEngKeyFunderBalance]: WalletName.KeyFunder,
+  [AlertType.HighUrgencyRelayerBalance]: WalletName.Relayer,
+};
+
+const alertIdMapping: Record<AlertTypeValue, string> = {
+  [AlertType.LowUrgencyKeyFunderBalance]: 'KiJNg6p4k',
+  [AlertType.LowUrgencyEngKeyFunderBalance]: 'temp',
+  [AlertType.HighUrgencyRelayerBalance]: 'be64gvjo8jvuod',
+};
+
+export const alertThresholdFileMapping: Record<AlertTypeValue, string> = {
+  [AlertType.LowUrgencyKeyFunderBalance]:
+    configFileNameMapping[BalanceThresholdConfig.LowUrgencyKeyFunderBalance],
+  [AlertType.LowUrgencyEngKeyFunderBalance]:
+    configFileNameMapping[BalanceThresholdConfig.LowUrgencyEngKeyFunderBalance],
+  [AlertType.HighUrgencyRelayerBalance]:
+    configFileNameMapping[BalanceThresholdConfig.HighUrgencyRelayerBalance],
+};
+
+export interface AlertRule {
+  uid: string;
+  title: string;
+  condition: string;
+  data: Array<{
+    refId: string;
+    queryType: string;
+    relativeTimeRange: {
+      from: number;
+      to: number;
+    };
+    datasourceUid: string;
+    model: {
+      expr: string;
+    };
+  }>;
+}
+
+export async function getGrafanaAlert(
+  alertType: AlertTypeValue,
+  saToken: string,
+) {
+  const alertUid = alertIdMapping[alertType];
+
+  try {
+    const response = await fetch(
+      `${GRAFANA_URL}/api/v1/provisioning/alert-rules/${alertUid}`,
+      {
+        headers: {
+          Authorization: `Bearer ${saToken}`,
+          'Content-Type': 'application/json',
+        },
+      },
+    );
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+
+    const data = (await response.json()) as AlertRule;
+
+    const queries = data.data.map((d) => d.model.expr);
+
+    return {
+      title: data.title,
+      queries,
+      rawData: data,
+    };
+  } catch (error) {
+    if (error instanceof Error) {
```

Comment:
maybe just a stylistic difference but imo a try/catch here is unnecessary just to log and throw again, this should be the consumer's problem. Generally I think too many try catches for scripts that don't need to be super tolerant of errors make for hard to read code